src/ext/serializer.c

```diff
@@ -173,32 +175,125 @@ int ddtrace_serialize_simple_array(zval *trace, zval *retval TSRMLS_DC) {
 }
 
 #if PHP_VERSION_ID < 70000
-#define ADD_ELEMENT_IF_PROP_TYPE(name, type)                                                                    \
-    do {                                                                                                        \
-        zval *prop =                                                                                            \
-            zend_read_property(ddtrace_ce_span_data, span->span_data, (name), sizeof((name)) - 1, 1 TSRMLS_CC); \
-        if (Z_TYPE_P(prop) == (type)) {                                                                         \
-            zval *value;                                                                                        \
-            ALLOC_ZVAL(value);                                                                                  \
-            INIT_PZVAL_COPY(value, prop);                                                                       \
-            zval_copy_ctor(value);                                                                              \
-            add_assoc_zval(el, (name), value);                                                                  \
-        }                                                                                                       \
-    } while (0);
+static zval *_read_span_property(zval *span_data, const char *name, size_t name_len TSRMLS_DC) {
+    return zend_read_property(ddtrace_ce_span_data, span_data, name, name_len, 1 TSRMLS_CC);
+}
+
+static void _add_assoc_zval(zval *el, const char *name, zval *prop) {
+    zval *value;
+    ALLOC_ZVAL(value);
+    INIT_PZVAL_COPY(value, prop);
+    zval_copy_ctor(value);
+    add_assoc_zval(el, name, value);
+}
+
+void _serialize_exception(zval *el, zval *meta, ddtrace_span_t *span TSRMLS_DC) {
+    zval *exception, *msg, *stack;
+    zend_uint class_name_len;
+    const char *class_name;
+    int dup;
+    exception = span->exception;
```

Comment:
Same comment as for `ver > 70000`.  Would it make sense to add null check here ?

Reply:
On PHP 7, the exception is an object pointer, not a zval, so it's definitely not needed on 7.

Reply:
Ah thanks for clarification